src/map.cpp

```diff
@@ -509,13 +509,20 @@ bool Map::isTileClear(uint16_t x, uint16_t y, uint8_t z, bool blockFloor /*= fal
 
 namespace {
 
-bool checkSteepLine(uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1, uint8_t z)
+bool checkSteepLine(uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1, uint8_t z, float modifier = 0)
 {
 	float dx = x1 - x0;
-	float slope = (dx == 0) ? 1 : (y1 - y0) / dx;
+	float slope = (dx == 0) ? 1 : std::abs((y1 - y0) / dx);
+	if (y1 < y0) {
+		slope *= -1;
+	}
 	float yi = y0 + slope;
+	if (modifier != 0) {
+		yi += modifier;
+	}
 
 	for (uint16_t x = x0 + 1; x < x1; ++x) {
+		// g_game.addMagicEffect(Position(std::floor(yi + 0.1),x,z), CONST_ME_POFF); // debugging
```

Comment:
What do these codes commented in a PR do to merge?

Reply:
Please, see the other comment with identical response.